### ISSUE TITLE
[sqlancer] execute sql error

### ISSUE THREAD

**Issue by wanpengfei-git** [StarRocks/starrocks#9943]
### Steps to reproduce the behavior (Required)
```
DROP DATABASE IF EXISTS g7;
CREATE DATABASE g7;
USE g7;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 VARCHAR(1) NOT NULL,c_0_1 DATE NOT NULL,c_0_2 DATETIME NOT NULL,c_0_3 ARRAY<INT> NOT NULL) PRIMARY KEY (c_0_0,c_0_1) PARTITION BY RANGE(c_0_1) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 175 day)) DISTRIBUTED BY HASH (c_0_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 VARCHAR(1) NOT NULL,c_1_1 DATE NOT NULL,c_1_2 BOOLEAN NOT NULL,c_1_3 BOOLEAN NOT NULL,c_1_4 BOOLEAN NOT NULL,c_1_5 VARCHAR(1) NOT NULL,c_1_6 VARCHAR(1) NOT NULL,c_1_7 DATE NOT NULL,c_1_8 DATE NOT NULL,c_1_9 BOOLEAN NOT NULL,c_1_10 BOOLEAN NOT NULL,c_1_11 BOOLEAN NOT NULL,c_1_12 TINYINT NOT NULL,c_1_13 DOUBLE NOT NULL, INDEX index1_c_1_5(`c_1_5`) USING BITMAP, INDEX index1_c_1_12(`c_1_12`) USING BITMAP) PRIMARY KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6,c_1_7,c_1_8,c_1_9,c_1_10,c_1_11,c_1_12) PARTITION BY RANGE(c_1_1) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 135 day)) DISTRIBUTED BY HASH (c_1_11,c_1_5,c_1_12,c_1_9,c_1_0,c_1_7,c_1_1,c_1_6) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t0 VALUES ('', '2010-01-08', '2010-01-18 08:00:14', []), ('', '2010-01-21', '2010-01-05 08:00:14', [-1221438665, -1221438665, -1624509585, -1624509585, 756618165, -1529841215, 1808487953, 1411409397, 1473167933, 58593736, 1568067930]), ('', '2010-01-14', '2010-01-09 08:00:14', [1808487953]);
INSERT INTO t0 VALUES ('', '2010-01-10', '2010-01-30 08:00:14', [-1624509585, -1221438665, -327345376, 1568067930, 2024092220, 1808487953, -1529841215, 1073441229]);
INSERT INTO t1 VALUES ('', '2010-01-13', false, false, true, '', '', '2010-01-14', '2010-01-06', false, false, false, 3, 1.808487953E9), ('', '2010-01-20', false, false, false, '', '', '2010-01-17', '2010-01-31', true, true, true, 2, 0.8768514187523772), ('', '2010-01-11', false, true, false, '', '', '2010-01-13', '2010-01-24', false, false, true, 0, 0.3050199037861643);
INSERT INTO t1 VALUES ('', '2009-12-31', true, false, false, '', '', '2010-01-14', '2010-01-16', true, true, true, 7, -3.27345376E8);
INSERT INTO t1 VALUES ('', '2010-01-04', true, false, true, '', '', '2010-01-13', '2010-01-27', true, true, true, 5, 5.8593736E7), ('', '2010-01-30', true, true, true, '', '', '2010-01-06', '2009-12-31', false, true, true, 7, 0.415523884951014);
INSERT INTO t1 VALUES ('', '2010-01-13', true, false, true, '', '', '2010-01-06', '2010-01-29', false, true, false, 3, 0.9374574808947632), ('', '2010-01-26', false, false, false, '', '', '2010-01-17', '2010-01-29', true, false, false, 0, 0.766988755150349);
INSERT INTO t1 VALUES ('', '2010-01-02', true, false, true, '', '', '2010-01-21', '2010-01-21', false, true, false, 4, 0.8795146165529256), ('', '2010-01-05', false, false, true, '', '', '2010-01-23', '2010-01-05', true, false, false, 4, 0.3611501128691532), ('', '2010-01-14', true, false, false, '', '', '2010-01-05', '2010-01-06', false, true, false, 0, 7.56618165E8), ('', '2010-01-17', false, false, true, '', '', '2010-01-20', '2010-01-23', false, false, true, 7, 0.274201560337494);
INSERT INTO t1 VALUES ('', '2010-01-20', true, false, true, '', '', '2010-01-21', '2010-01-14', true, false, false, 1, 0.510799782073904), ('', '2010-01-05', false, false, true, '', '', '2010-01-06', '2010-01-02', true, true, false, 3, -1.529841215E9);
INSERT INTO t0 VALUES ('', '2010-01-20', '2010-01-27 08:00:14', [1568067930, 8820451, -1730661090, 1159294253, 1568067930, 1916493375, -1624509585, -69785580, 1568067930, 1529634774, -1647438133, -1529841215, 1166314122, 1411409397]), ('', '2010-01-06', '2010-01-23 08:00:14', [1411409397, 153151871, 1302927427, -1128011492, 14874567, -785921554, 756618165, -1740674577, 153151871, 1967497905, 1159294253, -1701406668, 841933404]);
INSERT INTO t1 VALUES ('', '2010-01-11', true, true, false, '', '', '2010-01-30', '2010-01-24', false, true, true, 3, 0.766988755150349), ('', '2010-01-03', false, true, true, '', '', '2010-01-09', '2010-01-05', true, false, true, 7, 0.415523884951014);
DELETE FROM t1 WHERE c_1_8 IS NOT NULL AND c_1_11=true AND c_1_2=true AND c_1_3=false AND c_1_12=868059383 AND c_1_9=false;
INSERT INTO t0 VALUES ('', '2010-01-30', '2010-01-13 08:00:14', [-29938682, -1128011492, 282241740, 841933404, -583904049, 1643975625, 1916493375, 14874567, -1730661090, -1928991360, -1645286350, -1103836540]);
INSERT INTO t1 VALUES ('', '2010-01-10', false, true, true, '', '', '2010-01-12', '2010-01-21', true, false, false, 2, 1.643975625E9), ('', '2010-01-04', true, false, false, '', '', '2010-01-20', '2010-01-16', true, true, false, 4, 0.28283920290289655);
INSERT INTO t0 VALUES ('', '2010-01-23', '2010-01-13 08:00:14', []);
INSERT INTO t0 VALUES ('', '2010-01-31', '2010-01-02 08:00:14', [8820451, 1808487953]), ('', '2010-01-05', '2010-01-01 08:00:14', [2021106160, 1848054149, -1303646577, 343146281, -9223372036854775808, -1529841215, -970983599, 880921377, 170608169, -327345376, 61661060, 1302927427, -1730661090, 1480765903, 351677269]);
INSERT INTO t1 VALUES ('', '2010-01-18', true, true, false, '', '', '2010-01-15', '2010-01-05', false, true, true, 3, 0.8795146165529256);
INSERT INTO t0 VALUES ('', '2010-01-08', '2010-01-23 08:00:14', [398589777]);
SELECT t1_69.c_1_7, t1_69.c_1_10 FROM t0 AS t0_68, t1 AS t1_69 WHERE  ( (SELECT t1_69.c_1_3 FROM t1 AS t1_69 WHERE false)  ) < ( (SELECT t0_68.c_0_0 FROM t0 AS t0_68 WHERE (NOT ((((t1_69.c_1_11) IS NOT NULL)) IN (true) )))  )  GROUP BY t1_69.c_1_7, t1_69.c_1_10 UNION SELECT t1_69.c_1_7, t1_69.c_1_10 FROM t0 AS t0_68, t1 AS t1_69 WHERE (NOT ( ( (SELECT t1_69.c_1_3 FROM t1 AS t1_69 WHERE false)  ) < ( (SELECT t0_68.c_0_0 FROM t0 AS t0_68 WHERE (NOT ((((t1_69.c_1_11) IS NOT NULL)) IN (true) )))  ) )) GROUP BY t1_69.c_1_7, t1_69.c_1_10 UNION SELECT t1_69.c_1_7, t1_69.c_1_10 FROM t0 AS t0_68, t1 AS t1_69 WHERE (( ( (SELECT t1_69.c_1_3 FROM t1 AS t1_69 WHERE false)  ) < ( (SELECT t0_68.c_0_0 FROM t0 AS t0_68 WHERE (NOT ((((t1_69.c_1_11) IS NOT NULL)) IN (true) )))  ) ) IS NULL) GROUP BY t1_69.c_1_7, t1_69.c_1_10;

```
```
mysql> SELECT t1_69.c_1_7, t1_69.c_1_10 FROM t0 AS t0_68, t1 AS t1_69 WHERE  ( (SELECT t1_69.c_1_3 FROM t1 AS t1_69 WHERE false)  ) < ( (SELECT t0_68.c_0_0 FROM t0 AS t0_68 WHERE (NOT ((((t1_69.c_1_11) IS NOT NULL)) IN (true) )))  )  GROUP BY t1_69.c_1_7, t1_69.c_1_10 UNION SELECT t1_69.c_1_7, t1_69.c_1_10 FROM t0 AS t0_68, t1 AS t1_69 WHERE (NOT ( ( (SELECT t1_69.c_1_3 FROM t1 AS t1_69 WHERE false)  ) < ( (SELECT t0_68.c_0_0 FROM t0 AS t0_68 WHERE (NOT ((((t1_69.c_1_11) IS NOT NULL)) IN (true) )))  ) )) GROUP BY t1_69.c_1_7, t1_69.c_1_10 UNION SELECT t1_69.c_1_7, t1_69.c_1_10 FROM t0 AS t0_68, t1 AS t1_69 WHERE (( ( (SELECT t1_69.c_1_3 FROM t1 AS t1_69 WHERE false)  ) < ( (SELECT t0_68.c_0_0 FROM t0 AS t0_68 WHERE (NOT ((((t1_69.c_1_11) IS NOT NULL)) IN (true) )))  ) ) IS NULL) GROUP BY t1_69.c_1_7, t1_69.c_1_10;
ERROR 1064 (HY000): Expr evaluate meet error: assert_true failed due to false value
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - mysql> select current_version();
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE e8ba97d |
+----------------------+

**Comment by liuyehcf** [comment]
It is by design that scalar subquery may throw runtime error.

**Comment by liuyehcf** [comment]
Fixed by #9946
